geneate_dynein_conformation_graphs/format_inital.py

```diff
@@ -0,0 +1,272 @@
+        table_data[:len(doublet_data), i] = doublet_data
+        class_data[:len(doublet_data), i] = doublet_data
+
+    # Define a color map for the classes => PEET 
+    class_colors = {
+        1: '#0173b2',
+        2: '#de8f05',
+        3: '#029e73',
+        4: '#de8f05'
+    }
+
+    # Create a colormap for class values
+    colors = np.vectorize(class_colors.get)(class_data)
+
+    fig, ax = plt.subplots()
+
+    # Hide axes
+    ax.xaxis.set_visible(False) 
+    ax.yaxis.set_visible(False)
+    ax.set_frame_on(False)
+
+    # Create the table
+    table = ax.table(cellColours=colors, #cellText=table_data
+                    colLabels=[f'Doublet {i+1}' for i in range(9)],
+                    cellLoc='center',
+                    loc='center',
+                    colWidths=[0.07 for i in range(9)])
+
+    # Adjust the font size
+    table.auto_set_font_size(True)
+
+    # Scale the table
+    table.scale(0.6, 0.35)
+
+    # Figure creation 
+    fig_name = f"{foldername}_figure.png"
+    fig_path = f"{analysis_folder}/{foldername}/{fig_name}"
+    plt.savefig(fig_path, dpi = 600)
+
+    print(f"Succesfully made a figure of {foldername}")
+
+def prepare_metadata():
+
+    metadata = {
+            "ready4analysis": False,
+            "firstDoublet": {},
+            "clockDirection": {},
+            "bendDirection": {}
+    }
+
+    folder_names = [name for name in os.listdir(rawdata_folder)]
+
+    for folder in folder_names:
+        metadata["firstDoublet"][folder] = 0
+        metadata["clockDirection"][folder] = "undetermined" 
+        metadata["bendDirection"][folder] = "undetermined" 
+    
+    try:
+        with open(metadata_file, 'w') as file:
+            file.write("ready4analysis: false\n\n")
+            
+            file.write("firstDoublet:\n")
+            for key, value in metadata["firstDoublet"].items():
+                file.write(f"  {key}: {value}\n")
+            file.write("\n")
+            
+            file.write("clockDirection:\n")
+            for key, value in metadata["clockDirection"].items():
+                file.write(f"  {key}: {value}\n")
+            file.write("\n")
+            
+            file.write("bendDirection:\n")
+            for key, value in metadata["bendDirection"].items():
+                file.write(f"  {key}: {value}\n")
+        print(f"Metadata written to '{metadata_file}' successfully.")
+    except IOError as e:
+        print(f"Error writing to '{metadata_file}': {e}")
+
+def read_metadata():
+    try:
+        with open(metadata_file, 'r') as file:
+            metadata = yaml.safe_load(file)
+        return metadata
+    except IOError as e:
+        print(f"Error reading '{metadata_file}': {e}")
+        return None
+
+# Ensure metadata file exists
+if not os.path.exists(metadata_file):
+    prepare_metadata()
+
+# Read the metadata file in memory
+yaml_file = read_metadata()
+
+# If Ready for analysis is false, return error
+if not yaml_file['ready4analysis']:
```

Comment:
this will error out badly if `read_metadata()` returned `None`, you can catch that by using python's short-circuit logic (it evaluates `if` statements from left to right and returns as early as it can)
```suggestion
if yaml_file is None or not yaml_file['ready4analysis']:
```